sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/models/cas-supplier-response.dto.ts

```diff
@@ -0,0 +1,52 @@
+export class CASSupplierResponse {
+  items: CASSupplierResponseItem[];
+  hasMore: boolean;
+  limit: number;
+  offset: number;
+  count: number;
+  links: CASSupplierResponseItemLink[];
+}
+export class CASSupplierResponseItem {
+  suppliernumber: string;
+  suppliername: string;
+  subcategory: string;
+  sin: string;
+  providerid?: string;
+  businessnumber?: null;
+  status: string;
+  supplierprotected?: string | null;
+  standardindustryclassification?: string;
+  lastupdated: string;
+  supplieraddress: CASSupplierResponseItemAddress[];
+}
+
+export class CASSupplierResponseItemAddress {
+  suppliersitecode: string;
+  addressline1: string;
+  addressline2?: string;
+  addressline3?: string;
+  city: string;
+  province: string;
+  country: string;
+  postalcode: string;
+  emailaddress?: string;
+  accountnumber?: string;
+  branchnumber?: string;
+  banknumber?: string;
+  eftadvicepref?: string;
+  providerid?: string;
+  status: string;
+  siteprotected?: string;
+  lastupdated: string;
+}
+
+export class CASAuthDetails {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
```

Comment:
If my understanding is correct, the default token expiration will be 60 minutes of idle time which means we really will never need to cache it. Does it sound right?

_An idle token will expire after a specific number of seconds, typically 3600.  The expiry time may be different for each environment.  The expiry times are shared with all Web Services on the server and will not be changed without consideration of the impact to all web services._